identity/aziot-identityd/src/lib.rs

```diff
@@ -455,6 +460,29 @@ impl Api {
         Ok(())
     }
 
+    fn check_prev_provisioning(&self) -> bool {
+        let mut prev_settings_path = self.settings.homedir.clone();
+        prev_settings_path.push("prev_state");
+
+        if !prev_settings_path.exists() {
+            return true;
+        }
+
+        let prev_settings = if let Ok(settings) = crate::configext::load_file(&prev_settings_path) {
+            settings
+        } else {
+            return true;
+        };
+
+        if prev_settings.provisioning == self.settings.provisioning {
```

Comment:
This won't catch cases where key ID / cert ID are the same but the key / cert is different now.

We expect people to run `aziotctl system reprovision` after changing provisioning settings, right? Is there a reason to automatically reprovision here, especially since it might be confusing when it happens in some cases but not others?

Reply:
I was thinking that there's no scenario where someone would change the provisioning settings and not want Identity Service to reprovision. Having to run the `system reprovision` command just seems like an unnecessary extra step. True, this change doesn't catch all cases where the config changes, but I think it's better than nothing.